http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GKEInterpreter.scala

```diff
@@ -0,0 +1,390 @@
+import org.broadinstitute.dsde.workbench.google2.GKEModels._
+import org.broadinstitute.dsde.workbench.google2.KubernetesClusterNotFoundException
+import org.broadinstitute.dsde.workbench.google2.KubernetesModels.{
+  KubernetesNamespace,
+  KubernetesSecret,
+  KubernetesSecretType
+}
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.NamespaceName
+import org.broadinstitute.dsde.workbench.leonardo.config._
+import org.broadinstitute.dsde.workbench.leonardo.db.{
+  DbReference,
+  appErrorQuery,
+  kubernetesClusterQuery,
+  nodepoolQuery,
+  _
+}
+import org.broadinstitute.dsde.workbench.leonardo.http._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.AppNotFoundException
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.CreateAppMessage
+import org.broadinstitute.dsde.workbench.leonardo.util.{SetUpProjectNetworkParams, VPCAlgebra}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+final case class GKEInterpreterConfig(securityFiles: SecurityFilesConfig,
+                                      ingressConfig: IngressHelmConfig,
+                                      monitorConfig: AppMonitorConfig,
+                                      clusterConfig: KubernetesClusterConfig)
+class GKEInterpreter[F[_]: Parallel: ContextShift](
+  config: GKEInterpreterConfig,
+  vpcAlg: VPCAlgebra[F],
+  gkeService: org.broadinstitute.dsde.workbench.google2.GKEService[F],
+  kubeService: org.broadinstitute.dsde.workbench.google2.KubernetesService[F],
+  blocker: Blocker
+)(implicit val executionContext: ExecutionContext,
+  timer: Timer[F],
+  logger: StructuredLogger[F],
+  dbRef: DbReference[F],
+  F: ConcurrentEffect[F]) {
+
+  def createApp(msg: CreateAppMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val createApp = for {
+      ctx <- ev.ask
+      _ <- logger.info(s"beginning app creation for app ${msg.appId} | trace id : ${ctx.traceId}")
+      getAppOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(msg.project, msg.appName).transaction
+      getApp <- F.fromOption(getAppOpt, AppNotFoundException(msg.project, msg.appName, ctx.traceId))
+      gkeClusterId = getApp.cluster.getGkeClusterId
+      namespaceName = getApp.app.appResources.namespace.name
+
+      _ <- kubeService.createNamespace(gkeClusterId, KubernetesNamespace(namespaceName))
+      secrets <- getSecrets(namespaceName)
+      _ <- secrets.parTraverse(secret =>
+        kubeService.createSecret(gkeClusterId, KubernetesNamespace(namespaceName), secret)
+      )
+      _ <- logger.info(s"finished app creation for ${msg.appId}")
+      //TODO create svc accts
+      //TODO helm create ingress
+      //TODO helm create galaxy
+      _ <- appQuery.updateStatus(msg.appId, AppStatus.Running).transaction
+    } yield ()
+
+    createApp.onError {
+      case e =>
+        for {
+          now <- nowInstant
```

Comment:
Can we use the `now` from the `AppContext` passed to this method instead of generating a new one?